otherlibs/runtime_events/runtime_events_consumer.c

```diff
@@ -345,19 +354,31 @@ void caml_runtime_events_free_cursor(struct caml_runtime_events_cursor *cursor){
   if (cursor->cursor_open) {
     cursor->cursor_open = 0;
 #ifdef _WIN32
-    UnmapViewOfFile(cursor->metadata);
+    UnmapViewOfFile(cursor->map);
     CloseHandle(cursor->ring_file_handle);
     CloseHandle(cursor->ring_handle);
 #else
     /* This cast is necessary for compatibility with Illumos' non-POSIX
       mmap/munmap */
-    munmap((void*)cursor->metadata, cursor->ring_file_size_bytes);
+    munmap((void*)cursor->map, cursor->ring_file_size_bytes);
 #endif
     caml_stat_free(cursor->current_positions);
     caml_stat_free(cursor);
   }
 }
 
+static char* get_map_offset(struct caml_runtime_events_cursor *cursor,
+                            uint64_t offset, int domain_num, uint64_t len)
+{
+  uint64_t limit = cursor->ring_file_size_bytes;
+  if (offset >= limit)
+    return NULL;
+  offset += domain_num * len;
+  if (offset >= limit || offset + len > limit || offset + len < offset)
```

Comment:
You can use a simpler logic here:
```suggestion
  if (offset >= limit || len > limit - offset)
```

Reply:
Indeed, these are all unsigned values, so overflow should be well defined.